globus_action_provider_tools/flask/helpers.py

```diff
@@ -105,3 +117,80 @@ def blueprint_error_handler(exc: Exception) -> ViewReturn:
         ),
         500,
     )
+
+
+def validate_input(
+    request_json: Dict[str, Any], input_body_validator: ActionInputValidatorType
+) -> ActionRequest:
+    """
+    Ensures the incoming request conforms to the ActionRequest schema and
+    the user-defined ActionProvider input schema.
+    """
+    try:
+        action_request = ActionRequest(**request_json)
+    except ValidationError as ve:
+        raise BadActionRequest(ve.errors())
+
+    input_body_validator(action_request.body)
+    return action_request
+
+
+def get_input_body_validator(
```

Comment:
Just placing this at the top of the function: this looks nice and clean as well. Another key part to how this holds together and it looks really good.